routing/http/server/server.go

```diff
@@ -398,8 +409,13 @@ func (s *server) GetIPNS(w http.ResponseWriter, r *http.Request) {
 
 	record, err := s.svc.GetIPNS(r.Context(), name)
 	if err != nil {
-		writeErr(w, "GetIPNS", http.StatusInternalServerError, fmt.Errorf("delegate error: %w", err))
-		return
+		if errors.Is(err, routing.ErrNotFound) {
+			writeErr(w, "GetIPNS", http.StatusNotFound, fmt.Errorf("delegate error: %w", err))
+			return
+		} else {
+			writeErr(w, "GetIPNS", http.StatusInternalServerError, fmt.Errorf("delegate error: %w", err))
+			return
+		}
```

Comment:
:thought_balloon: I think if we return here, we won't set meaningful cache-control on these 404s.

Probably not a big deal for IPNS responses, as their rate and failure raite will be lower, but could we refactor IPNS not found handling to include explicit `cache-control` header? 

Maybe reuse `setCacheControl(w, maxAgeWithoutResults, maxStale)` from other endpoints, and call it inside of `writeErr` if code is NotFound?